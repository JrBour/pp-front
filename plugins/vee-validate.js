import { extend } from 'vee-validate'
import { required, email, max, min, numeric, alpha } from 'vee-validate/dist/rules'

extend('required', {
  ...required,
  message: 'Ce champ est requis'
})

extend('numeric', {
  ...numeric,
  message: 'Veuillez saisir des chiffres'
})

extend('alpha', {
  ...alpha,
  message: 'Veuillez ne saisir que des lettres'
})

extend('min', {
  ...min,
  validate(value, args) {
    return value.length >= args.length
  },
  params: ['length'],
  message: 'Ce champ doit contenir au moins {length} caracteres'
})

extend('max', {
  ...max,
  validate(value, args) {
    return value.length <= args.length
  },
  params: ['length'],
  message: 'Ce champ doit contenir au maximum {length} caracteres'
})

extend('email', {
  ...email,
  message: 'Email invalide'
})
