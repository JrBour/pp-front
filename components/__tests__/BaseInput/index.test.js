import { shallowMount } from '@vue/test-utils'
import { ValidationProvider } from 'vee-validate'
import BaseInput from '@/components/BaseInput'

describe('BaseInput', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(BaseInput, {
      propsData: {
        text: 'Field',
        name: 'name',
        onChange: jest.fn(),
        defaultValue: 'test',
        error: ''
      },
      stubs: {
        Loader: true,
        ValidationProvider
      }
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
