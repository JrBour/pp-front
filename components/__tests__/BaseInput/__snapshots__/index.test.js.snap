// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInput snapshot renders properly 1`] = `
<div class="input__wrapper">
  <div class="input__container"><input id="name" name="name" step="any" type="text" class=""> <label for="name" class="">
      Field <span>*</span></label></div>
  <p></p>
</div>
`;
